src/azure-cli/azure/cli/command_modules/synapse/_client_factory.py

```diff
@@ -3,6 +3,14 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+
+def cf_resource_groups(cli_ctx, subscription_id=None):
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
```

Comment:
could you import the client factory for resource module?
https://github.com/Azure/azure-cli/blob/a3b384eba388bc31003967769ede09a91c162571/src/azure-cli/azure/cli/command_modules/resource/_client_factory.py#L67-L68

Reply:
> could you import the client factory for resource module?
> https://github.com/Azure/azure-cli/blob/a3b384eba388bc31003967769ede09a91c162571/src/azure-cli/azure/cli/command_modules/resource/_client_factory.py#L67-L68

Hi @Juliehzl , I am afraid that if we import the `cf_resource_groups` from resource module there will be potential issue if there is some changes in resource module. Current status is using the client factory from azure.cli.core. core.

Reply:
but if we have some changes for get_mgmt_service_client, we need to change these two functions.
and resource module is owned by azure cli team and there should be safer.

Reply:
> but if we have some changes for get_mgmt_service_client, we need to change these two functions.
> and resource module is owned by azure cli team and there should be safer.

Ok, do you mean that when you update get_mgmt_service_client  you need to update all the reference? If all the services refer the resource module, it will be easy for you to maintain.  

Now I have removed this in client factory because this is for get resource group's location. Now the location has use the validator "get_default_location_from_resource_group", so we do not need this.  Thanks~